placementApp/views.py

```diff
@@ -22,6 +22,20 @@ class PositionViewSet(
     queryset = Position.objects.all()
     serializer_class = PositionSerializer
 
+class CompanyViewSet(
+    mixins.CreateModelMixin, mixins.UpdateModelMixin, mixins.ListModelMixin, mixins.RetrieveModelMixin, viewsets.GenericViewSet,
+):
+    permission_classes = (permissions.AllowAny,)
+    queryset = Company.objects.all()
+    serializer_class = CompanySerializer
+    def patch(self, request, pk):
```

Comment:
There is no need for a patch function the UpdateModelMixin handles both update and partial update through put and patch on its own so no need to define or overwrite it

Reply:
Patch request woudn't work it said Patch isn't allowed.

Reply:
Use ModelViewSet for Company.

Reply:
I execute the code after removing patch function and both put and patch works fine